source/common/signal/signal_action.h

```diff
@@ -52,7 +52,9 @@ class SignalAction : NonCopyable {
 public:
   SignalAction()
       : guard_size_(sysconf(_SC_PAGE_SIZE)),
-        altstack_size_(std::max(guard_size_ * 4, static_cast<size_t>(MINSIGSTKSZ))) {
+        altstack_size_(
+            std::max(guard_size_ * 4, (static_cast<size_t>(MINSIGSTKSZ) + guard_size_ - 1) /
+                                          guard_size_ * guard_size_)) {
```

Comment:
Maybe we can test this by define a static variable, then we override that in the test.